### PR TITLE
DOC: Fixes for SA04 errors in docstrings

### PR DESCRIPTION
**Pull request by suparnasnair**
- [x] Fixes for #28792 
- [x] tested through ./scripts/validate_docstrings.py

### CODE REVIEW

File: pandas/core/groupby/groupby.py
Comment (topper-123) [inline pandas/core/groupby/groupby.py]: The method should be ``bfill``, backfill doesn't exist on DataFrame/Series.

**Comment by gfyoung** [comment]
@suparnasnair : Thanks for the PR!  And welcome to `pandas` !

You're off to a good start, and I think as a next step, you should review your changes and condense them to just modifying the documentation.  Some of your changes are changing actual code, which is probably why there are test failures.

File: pandas/core/indexes/base.py
Comment (topper-123) [inline pandas/core/indexes/base.py]: "Return a Numpy array or Pandas  **ExtensionArray** of the Index" or similar

File: pandas/core/resample.py
Comment (topper-123) [inline pandas/core/resample.py]: Use type names ("str" instead of "string" etc).

File: pandas/core/resample.py
Comment (topper-123) [inline pandas/core/resample.py]: "Fill missing values using..." is IMO better (also in DataFrame|Series.fillna).

File: pandas/core/window/ewm.py
Comment (topper-123) [inline pandas/core/window/ewm.py]: IMO ending with " over the specified axis" makes the sentence heavy to read. I think it would say the same if that ending was just dropped?

Same goes for the doc strings below.

File: pandas/core/window/rolling.py
Comment (topper-123) [inline pandas/core/window/rolling.py]: "Rolling aggregate..." to distinguish it from the sentence below.

File: pandas/io/html.py
Comment (topper-123) [inline pandas/io/html.py]: ``read_csv`` can actually also read various non-comma-separated values by supplying a different value to the ``sep`` parameter. Can you adjust the sentence a bit.

**Review by topper-123 — CHANGES_REQUESTED** [review]

**Comment by topper-123** [comment]
Thanks. A few changes suggested, otherwise looks good.

I can see now that you took some of the strings from the referenced functions. In those cases you can just ignore my comments and let those be fixed seperately.

**Comment by suparnasnair** [comment]
Hi @topper-123, thanks for pointing out the corrections! I'm working on it now.

**Comment by suparnasnair** [comment]
Hi @topper-123 , I have made changes as per your review comments, please have a look.
I have few pep8 requirements to update as well, which I will be doing today.  Thanks!

**Comment by WillAyd** [comment]
@suparnasnair looks to be a merge conflict as we've had some changes to the groupby modules. Can you resolve?

**Comment by suparnasnair** [comment]
Sure @WillAyd. I'm on it.

**Comment by suparnasnair** [comment]
Hi @WillAyd, I have updated the code.  Now the only conflict is because of the extra line added due to the docstrings that I added.  Please review.

File: pandas/core/arrays/base.py
Comment (topper-123) [reply pandas/core/arrays/base.py]: Just call it the same as in the method "Take elements from an array.".

File: pandas/core/dtypes/cast.py
Comment (topper-123) [inline pandas/core/dtypes/cast.py]: The non-docstring changes need to go in a different PR.

File: pandas/core/groupby/groupby.py
Comment (topper-123) [reply pandas/core/groupby/groupby.py]: Just remove this line. The ``pad`` method doesn't actually exist.

File: pandas/core/indexes/base.py
Comment (topper-123) [reply pandas/core/indexes/base.py]: This still applies. Index.values does not necessarily return a numpy array, but can also return other types.

**Review by topper-123 — CHANGES_REQUESTED** [review]
A few remaining comments, see comments, else looks good

**Comment by suparnasnair** [comment]
Hi @topper-123 , I've updated based on your review.

**Comment by topper-123** [comment]
Single remaining issue is in pandas/core/dtypes/cast.py, those are not doc issues so should not be part of this PR. Else look good.

File: pandas/core/arrays/categorical.py
Comment (datapythonista) [inline pandas/core/arrays/categorical.py]: Is this from a merge, or is it intended?

File: pandas/core/arrays/categorical.py
Comment (datapythonista) [inline pandas/core/arrays/categorical.py]: same

File: pandas/core/dtypes/cast.py
Comment (datapythonista) [reply pandas/core/dtypes/cast.py]: same

**Review by datapythonista — COMMENTED** [review]
Thanks a lot for working on these @suparnasnair.

There are many changes to the code in this PR. I guess they are the result of merging master and fixing conflicts. You'll have to merge master again, fix the conflicts, and make sure you only have the intended changes in this PR, otherwise we can't merge it.

To merge master: `git fetch upstream` and `git merge upstream/master`

To check which changes you're proposing before pushing: `git diff upstream/master`

If you want to revert all the changes in your branch for a file:
- `git checkout origin/master pandas/core/dtypes/cast.py`
- `git checkout origin/master pandas/core/groupby/generic.py`

Hope it helps, let me know if you have any problem. Thanks!

**Comment by suparnasnair** [comment]
Hi @datapythonista, thanks for the comments. I'll merge and rebase the code once again and commit them. Working on it.

**Comment by suparnasnair** [comment]
Hi @datapythonista and @topper-123 , I have fixed all the merge issues.  Please let me know if there are any other updates required.  Thanks!

**Comment by datapythonista** [comment]
You've got changes to the code, not only docstrings. Are those intended? Have a look at the "Changes" tab to see what you're proposing, and if it's not what you intend please revert the changes from merges that didn't work as expected. I think I commented before how to do it, but let me know if you need help.

Thanks!

**Comment by FARVE14** [comment]
Hi @datapythonista , what I was doing is, merging my branch with master every time so as to update the changes made in those files to my branch.  But since, changes are made to those files mostly on a daily basis, the files in my branch always shows a conflict, since they are out-dated.  Could you please help me on what to do on that, if I perform a merge again with the master, this would definitely change the next day.  Or is it expected to not perform those merges?  

Thanks a lot in advance for the help, I'm just getting a hang of this slowly as I am new to this environment.

**Comment by datapythonista** [comment]
You've got lots of unrelated changes you need to undo. See my comments above. Those are the ones that are conflicting, since you shouldn't have them and they are quite big. Your intended changes won't conflict, nobody else is modifying those see also sections.

So, first thing make sure you don't have unintended changes. Then we can continue.

**Review by topper-123 — COMMENTED** [review]
The changes are all good, so if you can fix the merge errors that @datapythonista points out and rebase, this is good to pull in.

Fixing merge errors is one of the more difficult thing to get correct in Git, in my experience. If you feel more comfortable with that, you could also just split this PR into several smaller PR's. I've done that several times myself as it can be easier that becoming at expert at git.

**Comment by datapythonista** [comment]
I see that in this case you have many unintended changes. Those are probably the ones that are causing the conflicts. You can check the Files changes tab of this PR to see what they are. If they are not many, just change restore manually the original content.

For files that have lots of changes, and few updates to the see also made by you, you can `git checkout upstream/master pandas/core/dtypes/cast.py` (or the appropriate file name), and then reapply your changes again, should be faster.

Before doing the merge, run a `git diff upstream/master` and see what changes not to the docstrings are, and keep iterating until you just have your own changes to the see also version.

For the next contribution you make, make sure you work in small chunks of changes, will make your life easier.

**Comment by datapythonista** [comment]
@suparnasnair let me know when you're done, there are no conflicts, and there are no changes to the code, but only to the docstrings in https://github.com/pandas-dev/pandas/pull/28815/files and I'll have another look. Or if you need help.

**Comment by suparnasnair** [comment]
@datapythonista , I have for now removed all the changes I've made to those files. Currently there are no conflicts.  I will create separate pull requests for the changes I intended to make (docstrings) for these files.  Please have a look.

**Comment by jreback** [comment]
can you merge master

**Comment by WillAyd** [comment]
@suparnasnair looks like some merge conflicts have crept in - any chance you can fix those?

**Comment by datapythonista** [comment]
Closing as stale, @suparnasnair let us know if you want to finish this, and we'll reopen.

**Pull request closed without merging**
